cudamapper/tests/Test_CudamapperIndexGPUTwoIndices.cu

```diff
@@ -0,0 +1,1333 @@
+    expected_unique_representations.push_back(0b0010);
+    expected_first_occurrence_of_representations.push_back(2);
+    expected_representations.push_back(0b0100); // CA(2f1)
+    expected_positions_in_reads.push_back(2);
+    expected_read_ids.push_back(1);
+    expected_directions_of_reads.push_back(SketchElement::DirectionOfRepresentation::FORWARD);
+    expected_unique_representations.push_back(0b0100);
+    expected_first_occurrence_of_representations.push_back(3);
+    expected_representations.push_back(0b0101); // CC(1f1)
+    expected_positions_in_reads.push_back(1);
+    expected_read_ids.push_back(1);
+    expected_directions_of_reads.push_back(SketchElement::DirectionOfRepresentation::FORWARD);
+    expected_unique_representations.push_back(0b0101);
+    expected_first_occurrence_of_representations.push_back(4);
+    expected_representations.push_back(0b1001); // GC(0f1)
+    expected_positions_in_reads.push_back(0);
+    expected_read_ids.push_back(1);
+    expected_directions_of_reads.push_back(SketchElement::DirectionOfRepresentation::FORWARD);
+    expected_unique_representations.push_back(0b1001);
+    expected_first_occurrence_of_representations.push_back(5);
+
+    expected_first_occurrence_of_representations.push_back(6);
+
+    test_function(filename,
+                  1, // <- only take second read
+                  2,
+                  minimizer_size,
+                  window_size,
+                  expected_representations,
+                  expected_positions_in_reads,
+                  expected_read_ids,
+                  expected_directions_of_reads,
+                  expected_unique_representations,
+                  expected_first_occurrence_of_representations,
+                  expected_read_id_to_read_name,
+                  expected_read_id_to_read_length,
+                  1);
+}
+
+namespace details
+{
+namespace index_gpu_two_indices
+{
+// ************ Test find_first_occurrences_of_representations **************
+
+void test_find_first_occurrences_of_representations(const thrust::host_vector<representation_t>& representations_h,
+                                                    const thrust::host_vector<std::uint32_t>& expected_starting_index_of_each_representation_h)
+{
+    const thrust::device_vector<representation_t> representations_d(representations_h);
+
+    thrust::device_vector<std::uint32_t> starting_index_of_each_representation_d;
+    thrust::device_vector<representation_t> unique_representations_d;
+    find_first_occurrences_of_representations(unique_representations_d,
+                                              starting_index_of_each_representation_d,
+                                              representations_d);
+
+    const thrust::host_vector<std::uint32_t> starting_index_of_each_representation_h(starting_index_of_each_representation_d);
+
+    ASSERT_EQ(starting_index_of_each_representation_h.size(), expected_starting_index_of_each_representation_h.size());
```

Comment:
Do you test the unique representations somewhere?

Reply:
Good point! I checked it in `test_find_first_occurrences_of_representations_kernel`, but not `test_find_first_occurrences_of_representations`

Reply:
done